app/controllers/application_controller.rb

```diff
@@ -52,6 +44,7 @@ def set_locale
   end
 
   def store_request_id
+    Instrumentation.append_to_log(request_id: request.uuid)
```

Comment:
Ahem.  Meant to add this comment here....
@alan I think this sorts the problem, and possibly addresses @mloughran's concerns as well.  Let me know what you think.